### PR TITLE
:chore: Start ZKey downloading as soon as webapp is opened

### PR DESCRIPTION
**Pull request by wryonik**
Also add gmail input logic.

### CODE REVIEW

**Comment by Divide-By-0** [comment]
If you branch off of this repo instead, render can put a PR preview in the PR!

**Review by saleel — CHANGES_REQUESTED** [review]

File: packages/app/tests/App.test.tsx [hunk 1]
```diff
@@ -39,8 +39,9 @@ jest.mock("wagmi", () => {
 });
 
 describe("App", () => {
-  test("renders app", () => {
+  test("renders app", async () => {
     render(<App />);
+    await new Promise((r) => setTimeout(r, 1000));
```
Comment (saleel) [inline packages/app/tests/App.test.tsx:44]: Is this really needed?

File: packages/app/tests/e2e-ui.test.ts [hunk 2]
```diff
@@ -69,21 +72,20 @@ describe("App.js", () => {
...
     expect(ethValue).toBe(testEthAddress);
   });
 
-  it("should start with an enabled prove button and status should be 'not-started'", async () => {
-    await page.waitForSelector("[data-testid='status-not-started']");
+  it("should start with an enabled prove button and status should be 'downloading-proof-files'", async () => {
     const proveButtonSelector = "button[data-testid='prove-button']";
     const proveButtonIsDisabled = await page.$eval(proveButtonSelector, button => (button as HTMLButtonElement).disabled);
-    expect(proveButtonIsDisabled).toBe(false);
+    expect(proveButtonIsDisabled).toBe(true);
```
Comment (saleel) [inline packages/app/tests/e2e-ui.test.ts:85]: I think tests are failing here now in CI. Did they pass locally?

File: packages/app/src/pages/MainPage.tsx [hunk 3]
```diff
@@ -165,6 +218,36 @@ export const MainPage: React.FC<{}> = (props) => {
     }
   };
 
+  useEffect(() => {
+    const downloadZKey = async () => {
```
[the remark this answers is not in the document]
Comment (saleel) [reply packages/app/src/pages/MainPage.tsx:222]: Nvm, I saw its handled below.

**Review by saleel — APPROVED** [review]

**Pull request merged**